test/framework/taints.go

```diff
@@ -0,0 +1,92 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/executables"
+)
+
+const ownerAnnotation = "cluster.x-k8s.io/owner-name"
+
+func (e *ClusterE2ETest) VaidateWorkerNodeTaints() {
```

Comment:
No need to change, just starting the convo:
Should we make this more generic, like `VaidateWorkerNodes`? (I'm cool if that's divided into multiple sub validations)

The reason I'm suggesting this is because these mappings from eks-a to actual kube node objects is universal, independent from using taints or not. In fact these validations should pass even if all worker node groups lack taints. And once we have mapped eks-a api to nodes, adding more validations (labels, number of replicas...) becomes very easy. So I wonder if this is a validation worth running in all tests or at least other tests not taints related.

I know this discussion is not related to your work for taints and that's why I don't want this to be a blocker. But I think you wrote a very valuable piece of code and I would love for us to take advantage of it.

Reply:
totally. 

To confirm I understand -- when you say

> In fact these validations should pass even if all worker node groups lack taints

are you saying, we should be able to do a more universal node group validation that the wngc has created the proper nodes PERIOD (e.g. the node matches the wngc -- taints, labels, number present, etc)?  

Yes I think we should abstract this mapping out from this specific method to be re-used elsewhere. It was a pain to do and should be re-used. I'm going to do a bit of that when I push the CP taints e2e tests but we should move it forward more.

And also agree with you, and Joey mentioned this as well, that we should be running these kinds of validations in all our tests to confrim the nodes are right and not just isolate them to specific small sub-feautres (like taints or node lables)

Reply:
Yeah, exactly. Cool, let's continue this convo when you push the CP taints tests then.